llmfoundry/models/hf/hf_base.py

```diff
@@ -80,7 +80,14 @@ def __init__(
         additional_train_metrics: Optional[list] = None,
         additional_eval_metrics: Optional[list] = None,
         should_save_peft_only: bool = True,
+        attn_implementation: str = 'eager',
     ):
+        if use_flash_attention_2 and attn_implementation != 'flash_attention_2':
+            warnings.warn(
+                'use_flash_attention_2 is set, this will override attn_implementation',
```

Comment:
```suggestion
                'use_flash_attention_2 is set, this will override attn_implementation. Set attn_implementation to flash_attention_2 directly to remove this warning.',
```